src/sp_session.c

```diff
@@ -0,0 +1,157 @@
+#include "php_snuffleupagus.h"
+#include "ext/session/php_session.h"
+
+ZEND_DECLARE_MODULE_GLOBALS(snuffleupagus);
+
+#ifdef ZTS
+static ts_rsrc_id session_globals_id = 0;
+#define SESSION_G(v) ZEND_TSRMG(session_globals_id, php_ps_globals *, v)
+#ifdef COMPILE_DL_SESSION
+ZEND_TSRMLS_CACHE_EXTERN();
+#endif
+#else
+static php_ps_globals *session_globals = NULL;
+#define SESSION_G(v) (ps_globals.v)
+#endif
+
+static int (*previous_sessionRINIT)(INIT_FUNC_ARGS) = NULL;
+static ZEND_INI_MH((*old_OnUpdateSaveHandler)) = NULL;
+
+// PS_READ_ARGS => void **mod_data, zend_string *key, zend_string **val,
```

Comment:
Either trash this comment, or replace `PS_READ_ARGS` by the arguments in the function prototype.